packages/fmg-core/test/framework/transitions.js

```diff
@@ -0,0 +1,323 @@
+            toState.gameCounter = 45;
+            await assertRevert(validTransition(fromState, toState));
+        });
+    });
+
+
+    describe("propose -> accept", () => {
+        beforeEach(() => {
+            fromState = CountingGame.proposeState({ ...defaults, turnNum: 1, stateCount: 1 });
+            toState = CountingGame.acceptState({ ...defaults, turnNum: 2, stateCount: 0 });
+        });
+
+        it("allows a valid transition", async() => {
+            assert(await validTransition(fromState, toState));
+        });
+
+        it("rejects a transition where the turnNum doesn't increment", async () => {
+            toState.turnNum = fromState.turnNum;
+            await assertRevert(validTransition(fromState, toState));
+        });
+
+        it("rejects any transition where the channel changes", async () => {
+            toState.channel = otherChannel;
+            await assertRevert(validTransition(fromState, toState));
+        });
+
+        it("rejects a transition not from the last propose state", async() => {
+            fromState.stateCount = 0;
+            await assertRevert(validTransition(fromState, toState));
+        });
+
+        it("rejects a transition where the balances changes", async() => {
+            toState.resolution = otherResolution;
+            await assertRevert(validTransition(fromState, toState));
+        });
+
+        it("rejects a transition where the count doesn't reset", async() => {
+            toState.stateCount = 2;
+            await assertRevert(validTransition(fromState, toState));
+        });
+
+        it("rejects a transition where the position changes", async() => {
+            toState.gameCounter = 45;
+            await assertRevert(validTransition(fromState, toState));
+        });
+    });
+
+    describe("propose -> conclude", () => {
+        beforeEach(() => {
+            fromState = CountingGame.proposeState({ ...defaults, turnNum: 1, stateCount: 1 });
+            toState = CountingGame.concludeState({ ...defaults, turnNum: 2 });
+        });
+
+        it("allows a valid transition", async () => {
+            assert(await validTransition(fromState, toState));
+        });
+
+        it("rejects a transition where the turnNum doesn't increment", async () => {
+            toState.turnNum = fromState.turnNum;
+            await assertRevert(validTransition(fromState, toState));
+        });
+
+        it("rejects any transition where the channel changes", async () => {
+            toState.channel = otherChannel;
+            await assertRevert(validTransition(fromState, toState));
+        });
+
+        it("rejects a transition where the balances changes", async () => {
+            toState.resolution = otherResolution;
+            await assertRevert(validTransition(fromState, toState));
+        });
+
+        it("rejects a transition not from the last propose state", async () => {
```

Comment:
This is contrary to the paper, where there is always at least one `X -> CONCLUDE` transition.